h2o-algos/src/main/java/hex/glm/RegressionInfluenceDiagnosticsTasks.java

```diff
@@ -0,0 +1,321 @@
+        double oneOverVarEst = 1.0 / Math.sqrt(row2Array[_betaSize]);
+        for (int index = 0; index < _betaSize; index++)
+          betaDiff[index] = (_betas[index] - row2Array[index]) * oneOverVarEst * _oneOverSqrtXTXDiag[index];
+      }
+      for (int colIndex = 0; colIndex< _betaSize; colIndex++)  // write new beta to new chunk
+        nc[colIndex].addNum(betaDiff[colIndex]);
+    }
+    
+    private void readRow2Array(double[] row2Array, Chunk[] chks, int rInd, int nCol) {
+      for (int index=0; index<nCol; index++)
+        row2Array[index] = chks[index].atd(rInd);
+    }
+  }
+  
+  public static class ComputeNewBetaVarEstimatedGaussian extends MRTask<ComputeNewBetaVarEstimatedGaussian> {
+    final double[][] _cholInv;  // XTX inverse: store cholInv without redundant predictors, not scaled by parms._obj_reg
+    final double[] _xTransY;    // store XTY of full dataset
+    final double[] _xTransYReduced; // same as xTransY, but changed when there is redundant columns
+    final int _betaSize;
+    final int _reducedBetaSize;
+    final int _newChunkWidth;
+    final Job _j;
+    final DataInfo _dinfo;
+    final double[][] _xTx;  // not scaled by parms._obj_reg
+    final double _weightedNobs;
+    final double _sumRespSq;
+    final boolean _foundRedCols;
+    final double[] _stdErr; // used to tell which predict is redundant
+    
+    public ComputeNewBetaVarEstimatedGaussian(double[][] cholInv, double[] xTY, Job j, DataInfo dinfo, double[][] gram,
+                                              double nobs, double sumRespSq, double[] stdErr) {
+      _cholInv = cholInv;
+      _xTransYReduced = xTY; // if redundant columns present, is reduced
+      _betaSize = stdErr.length;
+      _reducedBetaSize = cholInv.length;
+      _foundRedCols = !(_betaSize == _reducedBetaSize);
+      _newChunkWidth = _betaSize+1; // last one is for estimated variance
+      _j = j;
+      _dinfo = dinfo;
+      _xTx = gram;
+      _weightedNobs = nobs-_reducedBetaSize; // intercept already included in gram/chol
+      _sumRespSq = sumRespSq; // YTY
+      _stdErr = stdErr;
+      _xTransY = new double[_betaSize];
+      if (_foundRedCols) { // shrink xTransY
+       int count=0;
+       for (int index=0; index<_betaSize; index++)
+         if (!Double.isNaN(stdErr[index]))
+           _xTransY[index] = _xTransYReduced[count++];
+      } else {
+        System.arraycopy(_xTransYReduced, 0, _xTransY, 0, _reducedBetaSize);
+      }
+    }
+
+    @Override
+    public void map(Chunk[] chks, NewChunk[] nc) {
+      if (isCancelled() || (_j != null && _j.stop_requested()))
+        return; // timeout
+      double[] newBeta = new double[_betaSize];
+      double[] newBetaRed = new double[_reducedBetaSize];
+      double[] row2Array = new double[_betaSize];
+      double[] row2ArrayRed = new double[_reducedBetaSize];
+      double[][] tDoubleArray1 = new double[_reducedBetaSize][_reducedBetaSize];
+      double[] tArray1 = new double[_betaSize];
```

Comment:
If I understand the code correctly the `tArray1` is just a temporary array with convenient size that is reused often. I would suggest to name it something like `tmpArray1`. Having just `t` as a prefix made me think there is something to do with matrix transpose.

Reply:
You are so deep in linear algebra that you will think t is for transpose.  I changed it to tmpArray